From aa9d54d214bf05d6833b37fdff6aae9bc67bce4f Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:24:18 -0400
Subject: [pci] Move pci_proc_attach_device() to pci_bus_add_device()

Message-id: <20140719042418.19438.61672.stgit@gir.stowe>
Patchwork-id: 85977
O-Subject: [RHEL7.1 PATCH 15/81] PCI: Move pci_proc_attach_device() to pci_bus_add_device()
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: ef37702eb3cae2ec35d56ddd4f096cd47fec3f0d

commit ef37702eb3cae2ec35d56ddd4f096cd47fec3f0d
Author: Yinghai Lu <yinghai@kernel.org>
Date:   Sat Nov 30 14:40:28 2013 -0800

    PCI: Move pci_proc_attach_device() to pci_bus_add_device()

    4f535093cf8f ("PCI: Put pci_dev in device tree as early as possible")
    moved pci_proc_attach_device() from pci_bus_add_device() to
    pci_device_add().

    This moves it back to pci_bus_add_device(), essentially reverting that
    part of 4f535093cf8f.  This makes it symmetric with pci_stop_dev(),
    where we call pci_proc_detach_device() and pci_remove_sysfs_dev_files()
    and set dev->is_added = 0.

    [bhelgaas: changelog, create sysfs then attach proc for symmetry]
    Signed-off-by: Yinghai Lu <yinghai@kernel.org>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/bus.c b/drivers/pci/bus.c
index fc1b740..a037d81 100644
--- a/drivers/pci/bus.c
+++ b/drivers/pci/bus.c
@@ -176,6 +176,7 @@ int pci_bus_add_device(struct pci_dev *dev)
   */
  pci_fixup_device(pci_fixup_final, dev);
  pci_create_sysfs_dev_files(dev);
+ pci_proc_attach_device(dev);
 
  dev->match_driver = true;
  retval = device_attach(&dev->dev);
diff --git a/drivers/pci/probe.c b/drivers/pci/probe.c
index d2f8a97..7bcc76d 100644
--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -1432,8 +1432,6 @@ void pci_device_add(struct pci_dev *dev, struct pci_bus *bus)
  dev->match_driver = false;
  ret = device_add(&dev->dev);
  WARN_ON(ret < 0);
-
- pci_proc_attach_device(dev);
 }
 
 struct pci_dev *__ref pci_scan_single_device(struct pci_bus *bus, int devfn)
-- 
1.7.1